packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -151,6 +157,28 @@ class DataStore implements IDataStore {
         return this.fluidDataStoreChannel.request(request);
     }
 
+    /**
+     * Exposes a handle to the data store's root object / entrypoint. Use this as the primary way of interacting with
+     * the data store, and only fall back to requesting the root object through the request pattern if this property
+     * or the handle within it are not defined.
+     */
+    public get IFluidLoadable(): IFluidLoadable | undefined {
+        // While we plumb entrypoints everywhere and this way of getting to the data store's entrypoint could still be
+        // undefined, we have to do some sleight-of-hand and return an object whose 'handle' property is not undefined
+        // so it matches the definition of the IFluidLoadable interface.
+        const maybeIFluidLoadable: IFluidDataStoreChannel & FluidObject<IProvideFluidLoadable>
```

Comment:
Please add entry point to IFluidDataStoreChannel not to have implicit casts like that.

Reply:
Leaving as a discussion point to talk through with @anthony-murphy . I don't have a particular problem with it, it'd just mean accepting some changes in the public API surface (additional properties on that interface that would break forward-compat), which we were trying to avoid altogether for now. But you've said before that `IFluidDataStoreChannel` is not supposed to be relevant for exernal devs, correct?

Reply:
It is not relevant for users of data stores. It's relevant for implementers of data stores.
You are right that would be breaking change. There are many ways to solve it. The best (I think) is the one where
1. You build a draft PR with "final" picture, including usage, for demonstration purposes and start breaking it into pieces once it has good final shape:
2 "main" gets mostly FYI notices in breaking.md, any new API in optional form, any deprecations.
3. "next" gets actual breaking changes, possible 2 versions after above (though we actively discussing dropping N-2 support to N-1 level, thus allowing users to immediately get their changes to next).

You can go with the path you are going (i.e. try to shove as many changes as possible into main - I do it all the time :)), but I'd still add it as optional to channel interface and put a breaking.md note that it is going to be required in next release. Otherwise, you are delaying it by one major version and making it more painful for yourself.

Reply:
There is a lot of comments around the typing in the PR, and i agree it is generally terrible, but it does match how our code defines the typing. 

I would have loved previous features leveraging handles to have improved the situation rather than just shoe horning things in, generally using unsafe `as` casts, but they didn't. So this is that state of the world, and i prefer to see how things are, so we actually see the hoops we force customers to jump through, or to do unsafe things with `as` casting.

Today things are so bad, i don't think we can even build a draft PR that represents the end state in any reasonable way, it would be huge, and touch tons of things. so i would not block this on the requirement. I think the current approach of minimizing surface area impact like is take here is the correct one. That allows us to get the pieces in places, and we drive the shape forward.